### PR TITLE
Fix docker build in python

### PR DESCRIPTION
**Pull request by komalali**
This PR builds off of the feedback from #160 and removes the async code from the implementation. 

Original PR: https://github.com/pulumi/pulumi-docker/pull/160

Resolves: https://github.com/pulumi/pulumi-docker/issues/164

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -68,6 +68,7 @@ install:: provider
 		yarn install --offline --production && \
 		(yarn unlink > /dev/null 2>&1 || true) && \
 		yarn link
+	cd ${PACKDIR}/python/bin && $(PIP) install --user -e .
```
Comment (lukehoban) [inline Makefile:71]: I don't think we do the equivalent of this in other repos - was there a reason you wanted to add it here?  If so, should we be consistent in other repos?

File: sdk/python/pulumi_docker/docker.py [hunk 2]
```diff
@@ -0,0 +1,635 @@
...
+    password: pulumi.Input[str]
+
+    def __init__(self, registry: pulumi.Input[str], username: pulumi.Input[str], password: pulumi.Input[str]):
+        self.registry = registry
+        self.username = username
+        self.password = password
+
+
+# CacheFrom may be used to specify build stages to use for the Docker build cache. The final image
+# is always implicitly included.
```
Comment (lukehoban) [inline sdk/python/pulumi_docker/docker.py:39]: I know this is unchanged from the original PR - but as a note possibly for follow-up, I don't believe this style of comments will show up in IDEs or doc generators.  I believe we need to use the same PEP 257 docstring style as in (e.g) https://github.com/pulumi/pulumi-docker/blob/master/sdk/python/pulumi_docker/container.py.  (note that these really aren't comments - they are API documentation)

File: Makefile (hunk 1 above)
Comment (stack72) [reply Makefile:71]: This happens in *some* repos

Comment (komalali) [reply Makefile:71]: This is from the original #160 PR, I didn't add it. I don't have a strong opinion here, except that I agree that we should be consistent across repos.

File: sdk/python/pulumi_docker/docker.py (hunk 2 above)
Comment (komalali) [reply sdk/python/pulumi_docker/docker.py]: Fixed to use the correct docstring style

**Comment by rienafairefr** [comment]
Thanks @komalali and others for the work on rescuing my port :-)

**Pull request merged**